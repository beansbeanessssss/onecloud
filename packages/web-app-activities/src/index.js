import translations from '../l10n/translations'
import Activities from './views/Activities.vue'
// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Activities'),
  id: 'activities',
  icon: 'activities',
  isFileEditor: false
}

const routes = [
  {
    path: '/',
    name: 'activities-all',
    component: Activities,
    meta: {
      title: $gettext('Users')
    }
  }
]

const navItems = []

export default {
  appInfo,
  routes,
  translations,
  navItems
}
