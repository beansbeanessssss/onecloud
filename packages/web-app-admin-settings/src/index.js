import translations from '../l10n/translations'
import Users from './views/Users.vue'
import Groups from './views/Groups.vue'
import Spaces from './views/Spaces.vue'
// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Administration Settings'),
  id: 'admin-settings',
  icon: 'settings-4',
  isFileEditor: false
}

const routes = [
  {
    path: '/',
    redirect: () => {
      const permissionManager = window.Vue.$permissionManager
      if (permissionManager.hasUserManagement()) {
        return { name: 'admin-settings-users' }
      }
      return { name: 'admin-settings-spaces' }
    }
  },
  {
    path: '/users',
    name: 'admin-settings-users',
    component: Users,
    meta: {
      authContext: 'user',
      title: $gettext('Users')
    }
  },
  {
    path: '/groups',
    name: 'admin-settings-groups',
    component: Groups,
    meta: {
      authContext: 'user',
      title: $gettext('Groups')
    }
  },
  {
    path: '/spaces',
    name: 'admin-settings-spaces',
    component: Spaces,
    meta: {
      authContext: 'user',
      title: $gettext('Spaces')
    }
  }
]

const navItems = [
  {
    name: $gettext('Users'),
    icon: 'user',
    route: {
      path: `/${appInfo.id}/users?`
    },
    enabled: () => {
      const permissionManager = window.Vue.$permissionManager
      return permissionManager.hasUserManagement()
    }
  },
  {
    name: $gettext('Groups'),
    icon: 'group-2',
    route: {
      path: `/${appInfo.id}/groups?`
    },
    enabled: () => {
      const permissionManager = window.Vue.$permissionManager
      return permissionManager.hasUserManagement()
    }
  },
  {
    name: $gettext('Spaces'),
    icon: 'layout-grid',
    route: {
      path: `/${appInfo.id}/spaces?`
    },
    enabled: () => {
      const permissionManager = window.Vue.$permissionManager
      return permissionManager.hasSpaceManagement()
    }
  }
]

export default {
  appInfo,
  routes,
  translations,
  navItems
}
