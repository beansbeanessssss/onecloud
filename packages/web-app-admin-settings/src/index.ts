import translations from '../l10n/translations.json'
import General from './views/General.vue'
import Users from './views/Users.vue'
import Groups from './views/Groups.vue'
import Spaces from './views/Spaces.vue'
// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Administration Settings'),
  id: 'admin-settings',
  icon: 'settings-4',
  isFileEditor: false
}

// FIXME: a better way to access this is needed
const permissionManager = () => (window as any).__$permissionManager

const routes = [
  {
    path: '/',
    redirect: () => {
      if (permissionManager().hasSystemManagement()) {
        return { name: 'admin-settings-general' }
      }
      return { name: 'admin-settings-spaces' }
    }
  },
  {
    path: '/general',
    name: 'admin-settings-general',
    component: General,
    meta: {
      authContext: 'user',
      title: $gettext('General')
    }
  },
  {
    path: '/users',
    name: 'admin-settings-users',
    component: Users,
    meta: {
      authContext: 'user',
      title: $gettext('Users')
    }
  },
  {
    path: '/groups',
    name: 'admin-settings-groups',
    component: Groups,
    meta: {
      authContext: 'user',
      title: $gettext('Groups')
    }
  },
  {
    path: '/spaces',
    name: 'admin-settings-spaces',
    component: Spaces,
    meta: {
      authContext: 'user',
      title: $gettext('Spaces')
    }
  }
]

const navItems = [
  {
    name: $gettext('General'),
    icon: 'settings-4',
    route: {
      path: `/${appInfo.id}/general?`
    },
    enabled: () => {
      return permissionManager().hasSystemManagement()
    }
  },
  {
    name: $gettext('Users'),
    icon: 'user',
    route: {
      path: `/${appInfo.id}/users?`
    },
    enabled: () => {
      return permissionManager().hasUserManagement()
    }
  },
  {
    name: $gettext('Groups'),
    icon: 'group-2',
    route: {
      path: `/${appInfo.id}/groups?`
    },
    enabled: () => {
      return permissionManager().hasUserManagement()
    }
  },
  {
    name: $gettext('Spaces'),
    icon: 'layout-grid',
    route: {
      path: `/${appInfo.id}/spaces?`
    },
    enabled: () => {
      return permissionManager().hasSpaceManagement()
    }
  }
]

export default {
  appInfo,
  routes,
  translations,
  navItems
}
