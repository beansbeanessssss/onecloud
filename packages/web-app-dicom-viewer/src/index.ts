import translations from '../l10n/translations.json'
import { AppWrapperRoute } from 'web-pkg/src/components/AppTemplates/AppWrapperRoute'
import DicomViewer from './DicomViewer.vue'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const routes = [
  {
    path: '/:driveAliasAndItem(.*)?',
    component: AppWrapperRoute(DicomViewer, {
      applicationId: 'dicom-viewer',
      urlForResourceOptions: {
        //disposition: 'inline'
      }
    }),
    name: 'dicom-viewer',
    meta: {
      authContext: 'hybrid',
      title: $gettext('DICOM Viewer'),
      patchCleanPath: true
    }
  }
]

const appInfo = {
  name: $gettext('DICOM Viewer'),
  id: 'dicom-viewer',
  icon: 'resource-type-medical',
  iconFillType: 'fill',
  iconColor: 'var(--oc-color-icon-medical)',
  extensions: [
    {
      extension: 'dcm',
      routeName: 'dicom-viewer'
    }
  ]
}

export default {
  appInfo,
  routes,
  translations
}
