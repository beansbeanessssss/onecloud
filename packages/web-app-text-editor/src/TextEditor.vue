<template>
  <div :class="showPreview ? 'oc-width-1-2' : 'oc-width-1-1'" class="oc-height-1-1">
    <oc-textarea
      id="text-editor-input"
      :model-value="currentContent"
      name="input"
      full-width
      label=""
      class="oc-height-1-1"
      :rows="20"
      :disabled="isReadOnly"
      @update:model-value="$emit('update:currentContent')"
    />
  </div>
  <div v-if="showPreview" class="oc-container oc-width-1-2">
    <!-- eslint-disable-next-line vue/no-v-html -->
    <div id="text-editor-preview" v-html="renderedMarkdown" />
  </div>
</template>

<script lang="ts">
import { marked } from 'marked'
import sanitizeHtml from 'sanitize-html'
import { computed, defineComponent, unref, PropType } from 'vue'
import { Resource } from 'web-client/src/helpers/resource/types'
import { AppConfigObject } from 'web-pkg/src/apps/types'

export default defineComponent({
  name: 'TextEditor',
  props: {
    applicationConfig: { type: Object as PropType<AppConfigObject>, required: true },
    currentContent: {
      type: String,
      required: true
    },
    serverContent: {
      type: String,
      required: true
    },
    isDirty: { type: Boolean, required: false },
    isReadOnly: { type: Boolean, required: false },
    resource: { type: Object as PropType<Resource>, required: true }
  },
  emits: ['update:currentContent'],
  setup(props) {
    const config = computed(() => {
      // TODO: Remove typecasting once vue-tsc has figured it out
      const { showPreviewOnlyMd = true } = props.applicationConfig as AppConfigObject
      return { showPreviewOnlyMd }
    })

    const showPreview = computed(() => {
      return props.resource.extension === 'md' || !unref(config).showPreviewOnlyMd
    })

    const renderedMarkdown = computed(() => {
      return props.currentContent && unref(showPreview)
        ? sanitizeHtml(marked(props.currentContent))
        : null
    })

    return {
      showPreview,
      renderedMarkdown
    }
  }
})
</script>

<style lang="scss">
#text-editor-preview {
  max-height: 100%;
  overflow-y: auto;
}
#text-editor-input {
  resize: vertical;
}
</style>
