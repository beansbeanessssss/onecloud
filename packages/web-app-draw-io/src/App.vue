<template>
  <main>
    <div v-if="loading" class="oc-position-center">
      <oc-spinner size="xlarge" />
      <p v-translate class="oc-invisible">Loading media</p>
    </div>
    <iframe
      v-else
      id="drawio-editor"
      ref="drawIoEditor"
      :src="iframeSource"
      :title="$gettext('Draw.io editor')"
    />
  </main>
</template>
<script lang="ts">
import { mapActions } from 'vuex'
import qs from 'qs'
import { DateTime } from 'luxon'
import { DavPermission, DavProperty } from 'web-pkg/src/constants'
import { useAppDefaults } from 'web-pkg/src/composables'
import { defineComponent } from '@vue/composition-api'
import { basename } from 'path'

export default defineComponent({
  name: 'DrawIoEditor',
  setup() {
    return {
      ...useAppDefaults({
        applicationId: 'draw-io'
      })
    }
  },
  data: () => ({
    loading: true,
    filePath: '',
    fileExtension: '',
    isReadOnly: null,
    currentETag: null
  }),
  computed: {
    config() {
      const {
        url = 'https://embed.diagrams.net',
        theme = 'minimal',
        autosave = false
      } = this.applicationConfig
      return { url, theme, autosave: autosave ? 1 : 0 }
    },
    iframeSource() {
      const query = qs.stringify({
        embed: 1,
        chrome: this.isReadOnly ? 0 : 1,
        picker: 0,
        stealth: 1,
        spin: 1,
        proto: 'json',
        ui: this.config.theme
      })

      return `${this.config.url}?${query}`
    }
  },
  created() {
    this.filePath = this.currentFileContext.path
    this.fileExtension = this.filePath.split('.').pop()
    this.checkPermissions()
    window.addEventListener('message', (event) => {
      if (event.data.length > 0) {
        const payload = JSON.parse(event.data)
        switch (payload.event) {
          case 'init':
            this.fileExtension === 'vsdx' ? this.importVisio() : this.load()
            break
          case 'autosave':
            this.save(payload, true)
            break
          case 'save':
            this.save(payload)
            break
          case 'exit':
            this.exit()
            break
        }
      }
    })
  },
  methods: {
    ...mapActions(['showMessage']),
    errorPopup(error) {
      this.showMessage({
        title: this.$gettext('An error occurred'),
        desc: error,
        status: 'danger'
      })
    },
    successPopup(msg) {
      this.showMessage({
        title: this.$gettext('The diagram was successfully saved'),
        desc: msg,
        status: 'success'
      })
    },
    errorNotification(error) {
      this.$refs.drawIoEditor.contentWindow.postMessage(
        JSON.stringify({
          action: 'status',
          message: error,
          modified: false
        }),
        '*'
      )
    },
    async checkPermissions() {
      try {
        const resource = await this.getFileInfo(this.currentFileContext, {
          davProperties: [DavProperty.Permissions]
        })
        this.isReadOnly = ![DavPermission.Updateable, DavPermission.FileUpdateable].some(
          (p) => (resource.permissions || '').indexOf(p) > -1
        )
        this.loading = false
      } catch (error) {
        this.errorPopup(error)
      }
    },
    load() {
      this.getFileContents(this.currentFileContext)
        .then((resp) => {
          this.currentETag = resp.headers.ETag
          this.$refs.drawIoEditor.contentWindow.postMessage(
            JSON.stringify({
              action: 'load',
              xml: resp.body,
              autosave: this.config.autosave
            }),
            '*'
          )
        })
        .catch((error) => {
          this.errorPopup(error)
        })
    },
    async importVisio() {
      const getDescription = () =>
        this.$gettextInterpolate(
          this.$gettext('The diagram will open as a new .drawio file: %{file}'),
          { file: basename(this.filePath) },
          true
        )
      // Change the working file after the import so the original file is not overwritten
      this.filePath += `_${this.getTimestamp()}.drawio`
      this.showMessage({
        title: this.$gettext('Diagram imported'),
        desc: getDescription()
      })
      this.getFileContents(this.currentFileContext, {
        responseType: 'arrayBuffer'
      })
        .then((resp) => {
          // Not setting `currentETag` on imports allows to create new files
          // otherwise the ETag comparison fails with a 412 during the autosave/save event
          // this.currentETag = resp.headers.get('etag')
          return resp.body
        })
        .then((arrayBuffer) => {
          const blob = new Blob([arrayBuffer], { type: 'application/vnd.visio' })
          const reader = new FileReader()
          reader.onloadend = () => {
            this.$refs.drawIoEditor.contentWindow.postMessage(
              JSON.stringify({
                action: 'load',
                xml: reader.result,
                autosave: this.config.autosave
              }),
              '*'
            )
          }
          reader.readAsDataURL(blob)
        })
        .catch((error) => {
          this.errorPopup(error)
        })
    },
    save(payload, auto = false) {
      this.putFileContents(this.currentFileContext, {
        content: payload.xml,
        previousEntityTag: this.currentETag
      })
        .then((resp) => {
          this.currentETag = resp.ETag

          const message = this.$gettext('File saved!')
          if (auto) {
            this.$refs.drawIoEditor.contentWindow.postMessage(
              JSON.stringify({
                action: 'status',
                message: message,
                modified: false
              }),
              '*'
            )
          } else {
            this.successPopup(message)
          }
        })
        .catch((error) => {
          const errorFunc = auto ? this.errorNotification : this.errorPopup
          switch (error.statusCode) {
            case 412:
              errorFunc(
                this.$gettext(
                  'This file was updated outside this window. Please refresh the page. All changes will be lost, so download a copy first.'
                )
              )
              break
            case 500:
              errorFunc(this.$gettext("Couldn't save. Error when contacting the server"))
              break
            case 401:
              errorFunc(this.$gettext("Saving error. You're not authorized to save this file"))
              break
            default:
              errorFunc(error.message || error)
          }
        })
    },
    exit() {
      window.close()
    },
    getTimestamp() {
      return DateTime.local().toFormat('YYYYMMDD[T]HHmmss')
    }
  }
})
</script>
<style scoped>
#drawio-editor {
  width: 100%;
  height: 100%;
  border: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
}
</style>
