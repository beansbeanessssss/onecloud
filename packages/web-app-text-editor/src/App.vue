<template>
  <AppWrapper
    v-slot="{ currentContent, ...slotArgs }: AppWrapperSlotArgs"
    application-id="text-editor"
  >
    <TextEditor v-model:current-content="currentContent.value" v-bind="slotArgs" />
  </AppWrapper>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import AppWrapper, { AppWrapperSlotArgs } from 'web-pkg/src/components/AppTemplates/AppWrapper.vue'
import TextEditor from './TextEditor.vue'

export default defineComponent({
  name: 'TextEditorApp',
  components: {
    AppWrapper,
    TextEditor
  }
})
</script>
