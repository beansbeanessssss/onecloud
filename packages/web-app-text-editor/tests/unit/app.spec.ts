import { AppConfigObject } from 'web-pkg/src/apps'
import { mount } from 'web-test-helpers'
import App from '../../src/App.vue'

jest.mock('web-pkg/src/composables/appDefaults')

describe('Text editor app', () => {
  it('shows the editor', async () => {
    const { wrapper } = getWrapper({
      applicationConfig: {}
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  // describe('preview', () => {
  //   it.each([
  //     { fileExtension: 'txt', showPreview: false },
  //     { fileExtension: 'js', showPreview: false },
  //     { fileExtension: 'php', showPreview: false },
  //     { fileExtension: 'json', showPreview: false },
  //     { fileExtension: 'xml', showPreview: false },
  //     { fileExtension: 'md', showPreview: true }
  //   ])('shows only for supported file types', async (data) => {
  //     const { wrapper } = getWrapper({ fileName: `file.${data.fileExtension}` })
  //     expect(wrapper.find('#text-editor-preview').exists()).toBe(data.showPreview)
  //   })
  // })
})

function getWrapper(props: { applicationConfig: AppConfigObject }) {
  return {
    wrapper: mount(App, {
      props
    })
  }
}
