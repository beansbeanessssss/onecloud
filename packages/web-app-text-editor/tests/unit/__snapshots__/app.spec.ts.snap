// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text editor app shows the editor 1`] = `
<div class="oc-width-1-1 oc-height-1-1">
  <!--v-if-->
  <!--v-if-->
</div>
`;
